sumologic/resource_sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -106,20 +109,26 @@ func getScheduleDefinitionSchemma() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			ForceNew: false,
 			Required: true,
-			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^\d{4}-\d{2}-\d{2}$`),
-				"start date in format of yyyy-mm-dd"),
+			ValidateFunc: validation.All(
+				validation.StringMatch(regexp.MustCompile(`^\d{4}-\d{2}-\d{2}$`),
+					"start date in format of yyyy-mm-dd"),
+				StringIsValidStartDate(),
+			),
 		},
 		"start_time": {
-			Type:         schema.TypeString,
-			Required:     true,
-			ForceNew:     false,
-			ValidateFunc: validation.StringLenBetween(5, 5),
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: false,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(5, 5),
+				StringIsValidStartTime(),
+			),
 		},
 		"duration": {
 			Type:         schema.TypeInt,
 			Required:     true,
 			ForceNew:     false,
-			ValidateFunc: validation.IntAtLeast(0),
+			ValidateFunc: validation.IntAtLeast(15),
```

Comment:
did we define any limit on the duration? 
does this allow negative value ?

Reply:
yes. we have a minimum duration of 15 minutes. 
no, it does not allow negative value.